vars/newmanExecute.groovy

```diff
@@ -1,201 +1,83 @@
-                node --version
-                npm --version
-            """
-            sh "NPM_CONFIG_PREFIX=~/.npm-global ${config.newmanInstallCommand}"
-            for(String collection : collectionList){
-                def collectionDisplayName = collection.toString().replace(File.separatorChar,(char)'_').tokenize('.').first()
-                // resolve templates
-                def command = GStringTemplateEngine.newInstance()
-                    .createTemplate(config.newmanRunCommand)
-                    .make([
-                        config: config.plus([newmanCollection: collection]),
-                        collectionDisplayName: collectionDisplayName
-                    ]).toString()
-                def command_secrets = ''
-                if(config.cfAppsWithSecrets){
-                    CloudFoundry cfUtils = new CloudFoundry(script);
-                    config.cfAppsWithSecrets.each { appName ->
-                        def xsuaaCredentials = cfUtils.getXsuaaCredentials(config.cloudFoundry.apiEndpoint,
-                                                    config.cloudFoundry.org,
-                                                    config.cloudFoundry.space,
-                                                    config.cloudFoundry.credentialsId,
-                                                    appName,
-                                                    config.verbose ? true : false ) //to avoid config.verbose as "null" if undefined in yaml and since function parameter boolean
-                        command_secrets += " --env-var ${appName}_clientid=${xsuaaCredentials.clientid}  --env-var ${appName}_clientsecret=${xsuaaCredentials.clientsecret}"
-                        echo "Exposing client id and secret for ${appName}: as ${appName}_clientid and ${appName}_clientsecret to newman"
-                    }
-                }
-
-                if(!config.failOnError) command += ' --suppress-exit-code'
-                try {
-                    if (config.cfAppsWithSecrets && command_secrets){
-                        echo "PATH=\$PATH:~/.npm-global/bin newman ${command} **env/secrets**"
-                        sh """
-                            set +x
-                            PATH=\$PATH:~/.npm-global/bin newman ${command} ${command_secrets}
-                        """
-                    }
-                    else{
-                        sh "PATH=\$PATH:~/.npm-global/bin newman ${command}"
-                    }
-                }
-                catch (e) {
-                    error "[${STEP_NAME}] ERROR: The execution of the newman tests failed, see the log for details."
-                }
-            }
+/**
+ * Runs code with secret environment variables and hides the values using Mask Passwords Plugin
+ *
+ * @param credentials - a list of maps with a 'var' and 'password' keys
+ * @param body - the code to run in
+ */
+def withSecretEnv(List<Map> credentials, Closure body) {
```

Comment:
This function uses Mask Passwords Plugin to mask secrets exposed to the step; since we can detect these secrets via environment variables, we are able to use log.RegisterSecret() to prevent logging secrets accidentally.

If there are no other ways to expose secrets, we _can_ remove the code.

Reply:
Removed due to unnecessary code